pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -127,10 +127,14 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 					// track more information about the dropped events.
 					conf := p.loadConfigurationFromCRD(ctxLog, k8sClient)
 
-					if reflect.DeepEqual(p.lastConfiguration.Get(), conf) {
+					confHash, err := hashstructure.Hash(conf, nil)
+					switch {
+					case err != nil:
+						logger.Error("Unable to hash the configuration")
+					case p.lastConfiguration.Get() == confHash:
 						logger.Debugf("Skipping Kubernetes event kind %T", event)
-					} else {
-						p.lastConfiguration.Set(conf)
+					default:
+						p.lastConfiguration.Set(confHash)
```

Comment:
Doing this means that the last configuration is not recoverable.

Did you want to have a `lastConfigurationHash`? and set the two (the config and the hash) if needed?

Reply:
We don't use `p.lastConfiguration.Get()` to get a configuration.